### PR TITLE
Adding `wait_for` support

### LINKED ISSUE #713: Explicit dependency relationships between electrons using wait for
## Acceptance Criteria

- There should be a new method `electron.wait_for(electron)` to establish more general [non-parameter] dependencies between electrons 
- This should be able to be applied for the use case of file transfer
- Add RTD & Docs

### PR DESCRIPTION
**Pull request by kessler-frost**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Resolves #713.

### CODE REVIEW

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]

File: covalent/_workflow/electron.py [hunk 1]
```diff
@@ -325,7 +325,19 @@ def __call__(self, *args, **kwargs) -> Union[Any, "Electron"]:
             return self.function(*args, **kwargs)
 
         if active_lattice.post_processing:
-            return active_lattice.electron_outputs.pop(0)
+
+            # This is to resolve `wait_for` calls during post processing time
+            id, output = active_lattice.electron_outputs[0]
+
+            for i, j, attr in active_lattice.transport_graph._graph.in_edges(id, data=True):
+                with open("mafile.txt", "a") as f:
```
[the remark this answers is not in the document]
Comment (santoshkumarradha) [reply covalent/_workflow/electron.py:333]: 😆

[the remark this answers is not in the document]
Comment (kessler-frost) [reply covalent/_workflow/electron.py:333]: 🥲 It wasn't intended to be here. Changing now.

File: covalent/_workflow/electron.py [hunk 2]
```diff
@@ -475,6 +487,46 @@ def to_electron_collection(**x):
...
+        # Just using list(electrons) will not work since we are overriding the __iter__
+        # method for an Electron which results in it essentially disappearing, thus using
+        # [electrons] to create the list if there's a single electron
+        electrons = [electrons] if isinstance(electrons, Electron) else list(electrons)
+
+        for el in electrons:
+            active_lattice.transport_graph.add_edge(
+                el.node_id,
+                self.node_id,
+                edge_name="!waiting_edge",
```
[the remark this answers is not in the document]
Comment (kessler-frost) [reply covalent/_workflow/electron.py:520]: Yeah, I agree, will do.

**Comment by AlejandroEsquivel** [comment]
@kessler-frost if documentation will be included in the PR - maybe change back to draft and mark as ready when done?

**Review by AlejandroEsquivel — APPROVED** [review]
Looks good @kessler-frost 👍

**Review by AlejandroEsquivel — DISMISSED** [review]

**Review by AlejandroEsquivel — APPROVED** [review]

**Pull request merged**